packages/platforms/browser/lib/auto-instrumentation/network-request-plugin.ts

```diff
@@ -17,23 +20,35 @@ export class NetworkRequestPlugin implements Plugin<BrowserConfiguration> {
   ) {}
 
   configure (configuration: InternalConfiguration<BrowserConfiguration>) {
+    this.logger = configuration.logger
+
     if (configuration.autoInstrumentNetworkRequests) {
       this.configEndpoint = configuration.endpoint
       this.xhrTracker.onStart(this.trackRequest)
       this.fetchTracker.onStart(this.trackRequest)
+      this.networkRequestCallback = configuration.networkRequestCallback
     }
   }
 
   private trackRequest = (startContext: RequestStartContext): RequestEndCallback | undefined => {
     if (!this.shouldTrackRequest(startContext)) return
 
+    const networkRequestInfo = this.networkRequestCallback({ url: startContext.url, type: startContext.type })
+
+    if (!networkRequestInfo) return
+
     const span = this.spanFactory.startSpan(
       `[HTTP]/${startContext.method.toUpperCase()}`,
       { startTime: startContext.startTime, makeCurrentContext: false }
     )
 
+    if (typeof networkRequestInfo.url === 'string') {
+      span.setAttribute('http.url', networkRequestInfo.url)
```

Comment:
I don't think this makes sense; is a network span without a URL ever going to be useful?

Reply:
The server also expects this to always be set for both network and resource spans, so we can't make it optional

Reply:
updated! if we return anything except a string we log a warning and don't create the span